app/src/main/java/com/odysee/app/ui/findcontent/AllContentFragment.java

```diff
@@ -466,6 +496,172 @@ public void onPause() {
         super.onPause();
     }
 
+    private void checkAppUpdater() {
+        if (BuildConfig.DEBUG) {
+            return;
+        }
+
+        Date now = new Date();
+
+        Context activity = getActivity();
+        // Check for a new version once in a day
+        if (activity != null && (now.getTime() - latestVersionCheck) > (24 * 3600 * 1000)) {
+            latestVersionCheck = now.getTime();
+
+            PackageManager pm = activity.getPackageManager();
+            try {
+                String installer;
+                if (Build.VERSION.SDK_INT > Build.VERSION_CODES.Q) {
+                    InstallSourceInfo installSourceInfo = pm.getInstallSourceInfo(activity.getPackageName());
+                    installer = installSourceInfo.getInstallingPackageName();
+                } else {
+                    //noinspection deprecation
+                    installer = pm.getInstallerPackageName(activity.getPackageName());
+                }
+
+                List<String> packagesNotAutoUpdating = Arrays.asList("com.google.android.feedback", "com.android.ending", "org.fdroid.fdroid");
+
+                if (installer == null || !packagesNotAutoUpdating.contains(installer)) {
+                    ((OdyseeApp)getActivity().getApplication()).getExecutor().submit(new Runnable() {
+                        @Override
+                        public void run() {
+                            checkRemoteLatestVersion(getView());
+                        }
+                    });
+                }
+            } catch (PackageManager.NameNotFoundException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    /**
+     * Checks for a new version on the remote server and then updates the app if a new one is available
+     * @param root the root view to show the snackbar
+     */
+    private void checkRemoteLatestVersion(View root) {
+        /* JSON file format:
+         * { latestVersionCode: integer,
+         *   url: string including https and package file }
+         */
+        Request.Builder builder = new Request.Builder();
+        builder = builder.url("http://apk.odysee.com/odysee-android/odysee-android-latest.json");
```

Comment:
I used a different domain because maybe they would prefer to use it in order to keep it separate from the non-native release. Anyway, I was also expecting to put there any server pointed at by the team.